hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/FSORepairCLI.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.debug;
+
+import org.apache.hadoop.hdds.cli.SubcommandWithParent;
+import org.kohsuke.MetaInfServices;
+import picocli.CommandLine;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Parser for scm.db file.
+ */
+@CommandLine.Command(
+    name = "fso-repair",
+    description = "Identify a disconnected FSO tree, and optionally mark " +
+        "unreachable entries for deletion. OM should be " +
+        "stopped while this tool is run. Information will be logged at " +
+        "INFO and DEBUG levels."
+)
+@MetaInfServices(SubcommandWithParent.class)
+public class FSORepairCLI implements Callable<Void>, SubcommandWithParent {
+
+  @CommandLine.Option(names = {"--db"},
+      required = true,
+      description = "Path to OM RocksDB")
+  private String dbPath;
+
+  @CommandLine.Option(names = {"--read-mode-only", "-r"},
+      required = true,
+      description =
+          "Mode to run the tool in. Read-mode will just log information about unreachable files or directories;" +
+              "otherwise the tool will move those files and directories to the deleted tables.",
+      defaultValue = "true")
+  private boolean readModeOnly;
+
+  @CommandLine.ParentCommand
+  private OzoneDebug parent;
+
+  @Override
+  public Void call() throws Exception {
+
+    try {
+      // TODO case insensitive enum options.
+      FSORepairTool repairTool = new FSORepairTool(dbPath, readModeOnly);
+      repairTool.run();
+    } catch (Exception ex) {
+      throw new IllegalArgumentException("FSO repair failed: " + ex.getMessage());
+    }
+
+    System.out.printf("FSO %s finished. See client logs for results.%n",
```

Comment:
Logging for CLI is off by default.

The tool should print all important pieces of information to std.out, and errors/warnings to std.err.  This allows users to see the results while avoiding chatty logs from some internal classes.

Print more detailed messages if `--verbose` is used.

Reply:
sure, added '--verbose' to command's option and moved this line of log into the code block when verbose = true.